### PR TITLE
PR for api calls

### PR DESCRIPTION
**Pull request by narharikale**
- added the following feature
- add a note.
- delete note.
- add to trash restore from the trash.
- archive, restore from the archive.

### CODE REVIEW

File: src/frontend/components/noteCard/noteCard.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    const foreverDelete = async(id) =>  {
+     
+        try {
+            const { data } = await axios.delete( `/api/notes/${id}` , {
+                headers:{
+                    authorization : isAuthorized.authtoken
+                }
+            } )
+            setNotesData(data.notes)    
+            console.log(data)
```
Comment (prathmeshjagtap) [inline src/frontend/components/noteCard/noteCard.jsx:28]: remove console.log

**Review by MiheerTamkhane — APPROVED** [review]
Code looks clean and working smooth.
Also the UI is looking very soothing 
Good work

File: src/frontend/components/noteCard/noteCard.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        }
+    };
+
+
+return (
+    <div className={`note-card ${color} `}>
+      {priority && <div class="badge-text">{priority}</div>}
+      <div className="note-card-header">
+        <div className="note-title"> {title} </div>
+        {/* <button className="icon-btn material-icons-outlined"> push_pin </button> */}
```
Comment (prathmeshjagtap) [inline src/frontend/components/noteCard/noteCard.jsx:54]: you can remove commented code

File: src/frontend/components/noteCard/noteCard.jsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+      <div
+        className="note-card-desc p-1"
+        dangerouslySetInnerHTML={{ __html: desc }}
+      />
+      <div>
+        {tags.map((tag, index) => (
+          <div key={index} className="badge-text">
+            {" "}
+            {tag}{" "}
```
Comment (prathmeshjagtap) [inline src/frontend/components/noteCard/noteCard.jsx:65]: remove{" "}

File: src/frontend/context/archive-context/ArchiveContext.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    const postArchive = async(id , note) =>  {
+        try {
+            const { data } = await axios.post( `/api/notes/archives/${id}` , {note} , {
+                headers:{
+                    authorization : isAuthorized.authtoken
+                }
+            } )
+            setArchiveData(data.archives)
+            setNotesData(data.notes)
+            console.log(data)
```
Comment (prathmeshjagtap) [inline src/frontend/context/archive-context/ArchiveContext.js:28]: remove console.log

File: src/frontend/context/archive-context/ArchiveContext.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    const restoreArchive = async(id , note) =>  {
+        try {
+            const { data } = await axios.post( `/api/archives/restore/${id}` , {note} , {
+                headers:{
+                    authorization : isAuthorized.authtoken
+                }
+            } )
+            setArchiveData(data.archives)
+            setNotesData(data.notes)
+            console.log(data)
```
Comment (prathmeshjagtap) [inline src/frontend/context/archive-context/ArchiveContext.js:45]: same here

**Review by prathmeshjagtap — APPROVED** [review]
Else Everything is good and well structured, Just remove console.logs and commented  code before merging

File: src/frontend/context/archive-context/ArchiveContext.js (hunk 4 above)
Comment (narharikale) [reply src/frontend/context/archive-context/ArchiveContext.js:28]: donr

File: src/frontend/components/noteCard/noteCard.jsx (hunk 2 above)
Comment (narharikale) [reply src/frontend/components/noteCard/noteCard.jsx:54]: yes

File: src/frontend/components/noteCard/noteCard.jsx (hunk 1 above)
Comment (narharikale) [reply src/frontend/components/noteCard/noteCard.jsx:28]: missed that !

**Pull request merged**